src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplateSupport.java

```diff
@@ -95,16 +96,21 @@ public CouchbaseDocument encodeEntity(final Object entityToEncode) {
 	}
 
 	@Override
-	public Object applyUpdatedCas(final Object entity, final long cas) {
+	public Object applyUpdatedCas(final Object entity, CouchbaseDocument converted, final long cas) {
```

Comment:
Added converted CouchbaseDocument to args as AfterSaveEvent takes a CouchbaseDocument.